ddtrace/contrib/loguru/patch.py

```diff
@@ -0,0 +1,129 @@
+import json
+
+import loguru
+from wrapt import wrap_function_wrapper as _w
+
+import ddtrace
+from ddtrace import config
+
+from ...internal.utils import get_argument_value
+from ..logging.constants import RECORD_ATTR_ENV
+from ..logging.constants import RECORD_ATTR_SERVICE
+from ..logging.constants import RECORD_ATTR_SPAN_ID
+from ..logging.constants import RECORD_ATTR_TRACE_ID
+from ..logging.constants import RECORD_ATTR_VALUE_EMPTY
+from ..logging.constants import RECORD_ATTR_VALUE_ZERO
+from ..logging.constants import RECORD_ATTR_VERSION
+from ..trace_utils import unwrap as _u
+
+
+config._add(
+    "loguru",
+    dict(),
+)
+
+
+def get_version():
+    # type: () -> str
+    return getattr(loguru, "__version__", "")
+
+
+def _tracer_injection(event_dict):
+
+    span = ddtrace.tracer.current_span()
+
+    trace_id = None
+    span_id = None
+    if span:
+        span_id = span.span_id
+        trace_id = span.trace_id
+        if config._128_bit_trace_id_enabled and not config._128_bit_trace_id_logging_enabled:
+            trace_id = span._trace_id_64bits
+
+    # add ids to structlog event dictionary
+    event_dict[RECORD_ATTR_TRACE_ID] = str(trace_id or RECORD_ATTR_VALUE_ZERO)
+    event_dict[RECORD_ATTR_SPAN_ID] = str(span_id or RECORD_ATTR_VALUE_ZERO)
+    # add the env, service, and version configured for the tracer
+    event_dict[RECORD_ATTR_ENV] = config.env or RECORD_ATTR_VALUE_EMPTY
+    event_dict[RECORD_ATTR_SERVICE] = config.service or RECORD_ATTR_VALUE_EMPTY
+    event_dict[RECORD_ATTR_VERSION] = config.version or RECORD_ATTR_VALUE_EMPTY
+
+    return event_dict
+
+
+def _w_add(func, instance, args, kwargs):
+    # patch logger to include datadog info before logging
+    instance.configure(patcher=lambda record: record.update(_tracer_injection(record)))
+    return func(*args, **kwargs)
+
+
+def _w_serialize(func, instance, args, kwargs):
```

Comment:
So when we log we have _w_add which is responsible for making sure the trace data is available in the record. Then when we serialize we have to make sure that that data is actually logged under the key we expect for backend correlation.

Reply:
Annoying that the log object is hard coded in the function.